libs/state/selections/src/lib/operators/select.ts

```diff
@@ -18,6 +25,43 @@ import { stateful } from './stateful';
 
 export function select<T>(): MonoTypeOperatorFunction<T>;
 
+/**
+ * @description
+ * Transform a slice of the state by providing keys and map function.
+ * Returns result of applying function to state slice as cached and distinct `Observable<R>`.
+ *
+ * @example
+ * // Project state slice
+ * const text$ = state$.pipe(
+ *   select(
+ *     ['query', 'results'],
+ *     ({ query, results }) => `${results.length} results found for "${query}"`
+ *   )
+ * );
+ *
+ * @return Observable<R>
+ */
+export function select<T extends object, K extends keyof T, R>(
+  keys: K[],
+  fn: (slice: PickSlice<T, K>) => R
+): OperatorFunction<T, R>;
```

Comment:
Not directly related to your PR but a general thought on the codebase, we shouldn't name generics with a single letter as it becomes hard to read and not really meaningful.